frontend/src/app/pages/exercises/exercise/shared/trainer-map-editor/trainer-map-editor.component.ts

```diff
@@ -82,4 +87,17 @@ export class TrainerMapEditorComponent {
             template: vehicleTemplate,
         });
     }
+
+    public importingTemplates = false;
+    public async importPartialExport(fileList: FileList) {
+        try {
+            this.importingTemplates = true;
+            openPartialImportOverwriteModal(
+                this.ngbModalService,
+                await fileList.item(0)?.text()
```

Comment:
This would result in a double validation, though, as the action will be validated later on.

Reply:
The idea is to validate (and migrate) it here **instead**.
This would not only improve the code (no confusing string that is passed to the modal) but also improve the UX as you get an error message if you try to import another JSON file before opening the modal.